x-pack/metricbeat/module/gcp/metrics/metrics_requester.go

```diff
@@ -139,111 +164,30 @@ func (r *metricsRequester) getFilterForMetric(serviceName, m string) string {
 		return f
 	}
 
-	switch serviceName {
-	case gcp.ServiceCompute:
-		if r.config.Region != "" && r.config.Zone != "" {
-			r.logger.Warnf("when region %s and zone %s config parameter "+
-				"both are provided, only use region", r.config.Regions, r.config.Zone)
-		}
+	var serviceLabel string
 
-		if r.config.Region != "" && len(r.config.Regions) != 0 {
-			r.logger.Warnf("when region %s and regions config parameters are both provided, use region", r.config.Region)
-		}
-
-		switch {
-		case r.config.Region != "":
-			f = fmt.Sprintf("%s AND %s = starts_with(\"%s\")", f, gcp.ComputeResourceLabel, strings.TrimSuffix(r.config.Region, "*"))
-		case r.config.Zone != "":
-			f = fmt.Sprintf("%s AND %s = starts_with(\"%s\")", f, gcp.ComputeResourceLabel, strings.TrimSuffix(r.config.Zone, "*"))
-		case len(r.config.Regions) != 0:
-			regionsFilter := r.buildRegionsFilter(r.config.Regions, gcp.ComputeResourceLabel)
-			f = fmt.Sprintf("%s AND %s", f, regionsFilter)
-		}
-	case gcp.ServiceGKE:
-		if r.config.Region != "" && r.config.Zone != "" {
-			r.logger.Warnf("when region %s and zone %s config parameter "+
-				"both are provided, only use region", r.config.Region, r.config.Zone)
-		}
-
-		switch {
-		case r.config.Region != "":
-			region := strings.TrimSuffix(r.config.Region, "*")
-			f = fmt.Sprintf("%s AND resource.label.location=starts_with(\"%s\")", f, region)
-		case r.config.Zone != "":
-			zone := strings.TrimSuffix(r.config.Zone, "*")
-			f = fmt.Sprintf("%s AND resource.label.location=starts_with(\"%s\")", f, zone)
-		case len(r.config.Regions) != 0:
-			regionsFilter := r.buildRegionsFilter(r.config.Regions, gcp.GKEResourceLabel)
-			f = fmt.Sprintf("%s AND %s", f, regionsFilter)
-		}
+	switch serviceName {
+	// Global GCP services that are not regional or zonal.
 	case gcp.ServicePubsub, gcp.ServiceLoadBalancing, gcp.ServiceCloudFunctions, gcp.ServiceFirestore:
 		return f
```

Comment:
nit: with directly `return f`, we missed the `r.logger.Debugf("ListTimeSeries API filter = %s", f)` debug logging here.

Reply:
I slightly refactored the code so that it always log that line in https://github.com/elastic/beats/pull/33560/commits/8549f0bff3964e012a7b3f56b2ea884ede2bdd71

What do you think?